### PR TITLE
Decoders

### PR DESCRIPTION
**Pull request by serpentk**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -45,36 +45,20 @@ async fn main() {
...
-            let rgb2 = Rgb::<u8>::from(yuv[1]);
-            let r = rgb2[0];
-            let g = rgb2[1];
-            let b = rgb2[2];
-            let alpha = 255;
-            let pixel = vec![r, g, b, alpha];
-            image_data[i * 2 + 1] = pixel.try_into().unwrap();
-        }
-        texture.update(&image);
+        // draw_texture(&texture, 0., 0., WHITE);
```
Comment (yashrk) [inline src/main.rs]: Let' s delete it.

File: src/main.rs [hunk 2]
```diff
@@ -10,32 +7,35 @@ use v4l::video::Capture;
...
+    let mut stream = Stream::with_buffers(&dev, Type::VideoCapture, BUFFER_COUNT)
         .expect("Failed to create buffer stream");
-    let mut image = Image::gen_image_color(WIDTH_U16, HEIGHT_U16, WHITE);
+    let mut image = Image::gen_image_color(fmt.width as u16, fmt.height as u16, WHITE);
     let texture = Texture2D::from_image(&image);
 
     let camera = Camera3D {
         position: vec3(-15., 15., -5.),
-        up: vec3(0., 0.001, 0.),
+        up: vec3(0., 0.1, 0.),
```
Comment (yashrk) [inline src/main.rs]: Is it important?

File: src/main.rs (hunk 1 above)
Comment (serpentk) [reply src/main.rs]: Done

File: src/main.rs (hunk 2 above)
Comment (serpentk) [reply src/main.rs]: No!

**Pull request merged**